devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java

```diff
@@ -411,123 +558,318 @@ public void execute() throws MojoExecutionException, MojoFailureException {
+        return false;
+    }
+
+    private boolean hasQuarkusDeploymentBom(List<Dependency> deps) {
+        if (deps == null) {
+            return false;
+        }
+        for (Dependency dep : deps) {
+            if (dep.getArtifact().getArtifactId().equals("quarkus-core-deployment")
+                    && dep.getArtifact().getGroupId().equals("io.quarkus")) {
+                // this is not a 100% accurate check but practically valid
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return true if the goal is executed in an existing project
+     */
+    private boolean isCurrentProjectExists() {
+        return currentProjectIsBaseDir ? project.getFile() != null
+                : Files.exists(basedir.toPath().resolve("pom.xml"));
+    }
+
+    private File getCurrentProjectPom() {
+        if (currentProjectIsBaseDir) {
+            return project.getFile() == null ? new File(project.getBasedir(), "pom.xml") : project.getFile();
         }
-        if (itestParentPath != null) {
-            generateItest(cfg, charset, model);
+        return new File(basedir, "pom.xml");
+    }
+
+    private Path getExtensionProjectBaseDir() {
+        if (currentProjectIsBaseDir) {
+            return project.getBasedir() == null ? basedir.toPath().resolve(artifactIdBase)
+                    : project.getBasedir().toPath().resolve(artifactIdBase);
         }
+        return new File(basedir, artifactIdBase).toPath();
+    }
+
+    private Path getExtensionRuntimeBaseDir() {
+        return getExtensionProjectBaseDir().resolve("runtime");
+    }
 
+    private Path getExtensionDeploymentBaseDir() {
+        return getExtensionProjectBaseDir().resolve("deployment");
     }
 
-    void generateItest(Configuration cfg, Charset charset, TemplateParams model)
-            throws MojoFailureException, MojoExecutionException, TemplateException {
-        final Path itestParentAbsPath = basedir.resolve(itestParentPath).toAbsolutePath();
+    void addModules(Path basePomXml, TemplateParams templateParams, Model basePom)
+            throws IOException, TemplateException, MojoFailureException, MojoExecutionException {
+        if (!basePom.getModules().contains(templateParams.artifactIdBase)) {
+            getLog().info(String.format("Adding module [%s] to [%s]", templateParams.artifactIdBase, basePomXml));
+            pomTransformer(basePomXml).transform(Transformation.addModule(templateParams.artifactIdBase));
+        }
+    }
+
+    private void generateExtensionProjects(Configuration cfg, TemplateParams templateParams)
+            throws IOException, TemplateException, MojoExecutionException {
+        evalTemplate(cfg, "parent-pom.xml", getExtensionProjectBaseDir().resolve("pom.xml"), templateParams);
+
+        Files.createDirectories(
+                getExtensionRuntimeBaseDir().resolve("src/main/java")
```

Comment:
I also want to see if this work on Windows, but CI should give us an answer soon enough :)